### ISSUE TITLE
ec2_vol msg: Cannot specify volume_size and either one of name or id

### ISSUE THREAD

**Issue by stolho** [ansible/ansible-modules-core#1612]
**Issue Type:**
Bug Report

**Ansible Version:**
ansible 1.9.1

**Ansible Configuration:**
No changes

**Environment:**
- Mac OS X 10.10.3
- Python 2.7.10
- The output of pip list:

``` shell
ansible (1.9.1)
boto (2.38.0)
ecdsa (0.13)
Jinja2 (2.7.3)
MarkupSafe (0.23)
paramiko (1.15.2)
pip (7.0.3)
pycrypto (2.6.1)
PyYAML (3.11)
setuptools (16.0)
wheel (0.24.0)
```

**Summary:**
Ansible fails to create a new EBS volume.

**Steps to reproduce:**
I have such code in my yaml file:

``` yaml
- name: Attach ebs volume to ec2 instance
  ec2_vol:
    instance: "i-xxxxxxxx"
    name: test-ebs-volume
    region: us-east-1
    volume_size: 40
```

And when I execute the ansible-playbook I get such error:

``` shell
TASK: [Attach ebs volume to ec2 instance] *************************************
failed: [localhost] => {"failed": true}
msg: Cannot specify volume_size and either one of name or id

FATAL: all hosts have already failed -- aborting
```

I was also trying to remove `instance: "i-xxxxxxxx"` in order just to create an ebs volume without attaching it to ec2 instance, but still get the same error.

I've found such code in my `ansible/modules/core/cloud/amazon/ec2_vol.py`:

``` python
# Delaying the checks until after the instance check allows us to get volume ids for existing volumes
# without needing to pass an unused volume_size
if not volume_size and not (id or name):
    module.fail_json(msg="You must specify an existing volume with id or name or a volume_size")

if volume_size and (id or name):
    module.fail_json(msg="Cannot specify volume_size and either one of name or id")
```

In the second `if` why should it fail if the user has specified volume_size and id or name? If I understand correct - if these things are specified - it should be enough to create the EBS volume?

**Comment by gregdek** [comment]
cc: @lwade

**Comment by leedm777** [comment]
Looks like something for this was merged into the devel branch (#729).

I'm guessing that it's a part of Ansible 2.0.

**Comment by wimnat** [comment]
@stolho can you check to see if my PR above resolves your issue?
